apps/fortuna/src/api/revelation.rs

```diff
@@ -20,7 +21,7 @@ use {
 /// Callers must pass the appropriate chain_id to ensure they fetch the correct random number.
 #[utoipa::path(
 get,
-path = "/v1/chains/{chain_id}/revelations/{sequence}",
+path = "/v1/chains/{chain_id}/revelations/{block_number}/{sequence}",
```

Comment:
this is a very awkward API for someone to call -- how would they know to get the block number also?

I think a better solution is to implement the cache of events for the explorer, and then use that to do the lookup from sequence number to block number.